Added max-width to govuk-select class

lucascumsille opened the pull request:
**Fixes:**
The multi-select has lost its maximum width at some point so e.g. if you’re on the dashboard for a council and pick a number of different wards, see how it gets bigger and bigger (or on a body admin page if you pick a number of areas to cover, etc)
![Screenshot 2024-08-14 at 22 34 13](https://github.com/user-attachments/assets/b3307d4c-88e0-4d6c-a856-922c843fb5bd)

**Preview**
This PR adds `max-width` to  the class `.govuk-select` 

https://github.com/user-attachments/assets/58239b8c-2fa9-4834-8e90-71f2bbf8c5b8

dracos reviewed (changes requested):
_report_lists.scss already has things like:
```
.multi-select-button {
    box-sizing: border-box;
    padding: flip(0.4em 1.5em 0.4em 0.6em, 0.4em 0.6em 0.4em 1.5em);
    vertical-align: top;
}
```
Can that now be removed/made more generic/used here instead?
Note the `flip` there to handle RTL languages as well.

Why is the `max-width` not 20em, as it was on .multi-select-button before it was removed - is there a reason for it to be 200px, where has that figure come from? Also, is that something we want on all selects, not just multi-select-button ones? I imagine there might be selects in use that are fine to be wider (e.g. the address one without JS is too narrow at 200px).

20px padding for the arrow seems a tiny bit too small:
![image](https://github.com/user-attachments/assets/807d802d-5cc7-4ff7-ba53-d37fb37ff5e2)

Lastly, I see _waste.scss includes a global `govuk-select` change which changes all govuk-selects, but only in those cobrands that have waste, which seems odd behaviour to have - should this be restricted to waste pages? If not, should it be in the generic CSS for all cobrands in some way? Just looking for an opportunity as always to simplify the CSS, not just add more lines on top :)

lucascumsille commented:
@dracos  I was planning to do the tidy up and simplification [here](https://github.com/mysociety/societyworks/milestone/99), ideally when we finish with that milestone all the selects and multi-selects should have the govuk styling so I'm sure we can simplify the CSS. In this sprint I'm not sure I have the time to do all the above + testing. So I'll modify this PR, to fix the bug we currently have, I'll create a ticket to tackle your comment.

lucascumsille commented:
@dracos I ended up moving `padding: flip(0.4em 1.5em 0.4em 0.6em, 0.4em 0.6em 0.4em 1.5em);` to `.govuk-select` to make it more generic. For testing I created a `input.govuk-select` nested juts inside a container to see if it would cause any issue:

<img width="1283" alt="Screenshot 2024-08-15 at 12 02 11" src="https://github.com/user-attachments/assets/2c3c5acc-2976-4a61-98ad-bfeaff7772fc">

This is the report page
<img width="474" alt="Screenshot 2024-08-15 at 12 02 48" src="https://github.com/user-attachments/assets/12c8e351-a271-4c7f-971a-54ef6986f971">

I added one commit `c638247` to make the height of all elements nested inside `.dashboard-filters` similar.  The overrides in the` line-height` and `height that we added a few months ago for the `.dashboard-filters .multi-select-button` elements ` was causing an issue with the new generic `padding: flip(0.4em 1.5em 0.4em 0.6em, 0.4em 0.6em 0.4em 1.5em);`, so I deleted them. Another solution would have been just to give the `govuk-select` to "Report state", but we still haven't implemented a styling for the `input[date]`, so 'Start date' and 'End date' would have still needed some extra `height` so not much of a solution at the end. I'll add the styling for the `input[date]` one to the milestone above.
 
**Preview c638247**

<img width="1283" alt="Screenshot 2024-08-15 at 11 42 18" src="https://github.com/user-attachments/assets/e1c9671e-7c63-40bd-983e-c0072e53c91a">

**Preview without c638247**

<img width="1283" alt="Screenshot 2024-08-15 at 11 36 32" src="https://github.com/user-attachments/assets/9d9a80bd-83e4-4ff1-8719-104511f822d9">

Let me know what you think =)

lucascumsille commented:
By the way the ` max-width: 10em;` prevents any second line jumping, so no matter how many selections are made, they "Look up" button doesn't jump to a second line.

dracos commented on web/cobrands/sass/_dashboard.scss:
```diff
@@ -297,8 +297,12 @@
+
+    .form-control {
+        // To make them have the same height as the .multi-select-button
+        height: 3.3em;
```
This is fine - something else bit odd to think about later, as we now have select.form-control { height: 2.2em }, .filters .form-control { height: 2.2em }, .report-list-filters .form-control { height: 1.8em } and now with this .dashboard-filters .form-control { height: 3.3em }. And the filters on the heatmap page and the triage page do not look as nice as the public filters, I don't think they were updated to match the public around page.

dracos reviewed (approved):
Fixes the issue, thanks.

As an example of my comment, for later, here's the Bromley heatmap filters at present:
![image](https://github.com/user-attachments/assets/582d2750-0fe5-4b3a-add3-4a4ad307d9af)
(I think the public filters changed the definition of what .report-list-filters was used for, so other users of that class have gone a little wrong.)
